src/Servers/ShellServer.cs

```diff
@@ -67,9 +67,9 @@ IShellRouter router
             this.provider = provider;
             this.router = router;
 
-            router.RegisterHandler("kernel_info_request", async message => KernelInfoRequest?.Invoke(message));
+            router.RegisterHandler("kernel_info_request", async message => OnKernelInfoRequest(message));
```

Comment:
It looks like you are replacing invocations of events with plain method calls. The OnShutdownRequest method that is called will do the invocation of the event it replaced, ensuring that any registered events handlers get invoked, but the OnKernelInfoRequest doesn't seem to do the same. Is there a risk here that KernelInfoRequest event handles don't get invoked when they are supposed to, or is this intentional?

Reply:
Looks like something went wrong in merging from `main`, the event should definitely be called in `OnKernelInfoRequest`. Thanks for catching that!